java/src/com/ibm/streamsx/topology/generator/spl/GraphUtilities.java

```diff
@@ -0,0 +1,310 @@
+package com.ibm.streamsx.topology.generator.spl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import com.ibm.json.java.JSONArray;
+import com.ibm.json.java.JSONObject;
+import com.ibm.streamsx.topology.function7.Consumer;
+
+public class GraphUtilities {
+    public static ArrayList<JSONObject> findStarts(JSONObject graph) {
+        ArrayList<JSONObject> starts = new ArrayList<JSONObject>();
+        JSONArray ops = (JSONArray) graph.get("operators");
+        for (int k = 0; k < ops.size(); k++) {
+            JSONObject op = (JSONObject) (ops.get(k));
+            JSONArray inputs = (JSONArray) op.get("inputs");
+            if (inputs == null || inputs.size() == 0) {
+                if (((JSONObject) ops.get(k)).get("name") != null
+                        && !((String) ((JSONObject) ops.get(k)).get("name"))
+                                .startsWith("$")) {
+                    starts.add((JSONObject) ops.get(k));
+                }
+            }
+        }
+        return starts;
+    }
+
+    public static ArrayList<JSONObject> findOperatorByKind(String opKind,
+            JSONObject graph) {
+        ArrayList<JSONObject> lowLatencyStarts = new ArrayList<JSONObject>();
+        JSONArray ops = (JSONArray) graph.get("operators");
+        for (int k = 0; k < ops.size(); k++) {
+            JSONObject op = (JSONObject) (ops.get(k));
+            String kind = (String) op.get("kind");
+            if (kind != null && kind.equals(opKind)) {
+                lowLatencyStarts.add(op);
+            }
+        }
+        return lowLatencyStarts;
+    }
+
+    // Get the children of the operator. Should probably be re-worked.
```

Comment:
Need some more detailed comments on what is returned, all downstream operators or only immediate? Any special handling for marker operators? Same for getParent.

Maybe consider chnaging the names to up/downstream.